### DISCUSSION TITLE
Failed to retrieve backup files: scp error (incorrect backup file name?)

### DISCUSSION THREAD

**Discussion by jmaciak** [eduardogsilva/routerfleet#50] [General]
**Description:**

backup fails with message:

_Failed to retrieve backup files: scp error: routerfleet-backup-2-instant-2024-11-25_13-28.rsc: no such file or directory!_

On the router, I can see file:

_routerfleet-backup-2-instant-2024-11-25_13-28.backup_

Is there something misconfigured on my side?

Thank you!

**Comment by eduardogsilva** [comment]
Hello @jmaciak 

This is a strange situation, I would say that "Probably" routerfleet executed the backup command, and tried to fetch the backup file before the router finished to write the file.

This can happen  for a lot of reasons, I will just suggest some reasons that I could think of.
1- Maybe a very slow storage on your router.
2- high cpu usage
3- large backup file

My suggestion would be to increase the "Retrieve interval" setting on your router backup profile.
This is the time that routerfleet will wait from the backup execution until the retrieve process.

There's no way to check if the router successfully completed the backup, so waiting a longer time until retrieval, should do the trick.

Cheers

**Reply by jmaciak** [reply]
Dear Eduardo,

thank you for your reply. I tried to diagnose the issue further and came with the following solution/observation:

the user performing backup (on Mikrotik) needs to have "write" permission. Without this permission, the user can create binary backup, but can't create rsc file (not sure why, maybe because write permission is required to write the file). And since the rsc file is being downloaded first, the download fails.

So it was a mistake on my side. After adding the write permission to the user performing backup, everything works as expected.

Thank you!

Juraj
